File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int arr[], size_t arr_len) {
    int* answer = NULL;
    int index = 0;
    for (int i = 0; i < arr_len; i++) {
        if (index == 0) {
            answer = realloc(answer, sizeof(int) * (index+1));
            answer[index++] = arr[i];
        } else if (answer[index-1] == arr[i]) {
            answer = realloc(answer, sizeof(int) * (index));
            index--;
        } else if (answer[index-1] != arr[i]) {
            answer = realloc(answer, sizeof(int) * (index+1));
            answer[index++] = arr[i];
        }
    }
    if (index == 0) {
        answer[0] = -1;
    }
    return answer;
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.js
function solution(arr) {
    var answer = [];
    for (let i in arr) {
        if (answer == []) {
            answer.push(arr[i]);
        } else if (arr[i] == answer[answer.length-1]) {
            answer.pop();
        } else if (arr[i] != answer[answer.length-1]) {
            answer.push(arr[i]);
        }
    }
    if (answer.length == 0) {
        answer.push(-1);
    }
    return answer;
}